src/backend/Attributable.cpp

```diff
@@ -400,23 +487,38 @@ void Attributable::readAttributes(ReadMode mode)
             guardUnitDimension(att, a.get<std::vector<long double> >());
             break;
         case DT::VEC_CFLOAT:
-            setAttribute(att, a.get<std::vector<std::complex<float> > >());
+            setAttributeImpl(
+                att,
+                a.get<std::vector<std::complex<float> > >(),
+                internal::SetAttributeMode::WhileReadingAttributes);
             break;
         case DT::VEC_CDOUBLE:
-            setAttribute(att, a.get<std::vector<std::complex<double> > >());
+            setAttributeImpl(
+                att,
+                a.get<std::vector<std::complex<double> > >(),
+                internal::SetAttributeMode::WhileReadingAttributes);
             break;
         case DT::VEC_CLONG_DOUBLE:
             setAttribute(
                 att, a.get<std::vector<std::complex<long double> > >());
```

Comment:
Is this one missing?
```suggestion
            setAttributeImpl(
                att,
                a.get<std::vector<std::complex<long double> > >(),
                internal::SetAttributeMode::WhileReadingAttributes);
```

Reply:
Added to last commit.

Reply:
Looks like it, good catch